src/Razor/src/Microsoft.VisualStudio.Editor.Razor/DefaultVisualStudioRazorParser.cs

```diff
@@ -394,14 +418,19 @@ internal void OnDocumentStructureChanged(object state)
                 return;
             }
 
-            _latestChangeReference = null;
-            _codeDocument = backgroundParserArgs.CodeDocument;
-            _snapshot = backgroundParserArgs.ChangeReference.Snapshot;
-            _partialParser = new RazorSyntaxTreePartialParser(CodeDocument.GetSyntaxTree());
+            lock (GetCodeDocumentLock)
+            {
+                _latestChangeReference = null;
+                _codeDocument = backgroundParserArgs.CodeDocument;
+                _snapshot = backgroundParserArgs.ChangeReference.Snapshot;
+                _partialParser = new RazorSyntaxTreePartialParser(CodeDocument.GetSyntaxTree());
+                _codeDocumentTaskCompletionSource?.SetResult(_codeDocument);
```

Comment:
Why does L426 use the property for `CodeDocument` and L427 use the field?

Reply:
No reason in particular, just an oversight.